onnxscript/rewriter/ort_fusions/shape_optimization.py

```diff
@@ -37,11 +38,22 @@ def check(self, context, dim0, dim1, dim2, dim3, final_shape, **_) -> bool:
             return False
         if "start" in shape_node.attributes:
             start_attr = shape_node.attributes["start"]
-            return isinstance(start_attr, ir.Attr) and start_attr.value == 0
+            if not (isinstance(start_attr, ir.Attr) and start_attr.value == 0):
+                return False
+        self._start_val = _ir_utils.get_singleton_value(start)
+        self._end_val = _ir_utils.get_singleton_value(end)
+        if self._start_val is None or self._end_val is None:
+            return False
         return True
 
-    def rewrite(self, op, dim1, **_):
-        return dim1
+    def rewrite(self, op, dim0, dim1, dim2, dim3, **_):
+        transposed_dims = [dim0, dim2, dim1, dim3]
+        sliced_result = transposed_dims[self._start_val : self._end_val]
```

Comment:
Looks like the lines are uncovered by tests

Reply:
Will add unit tests